packages/web-api-scan-runner/src/runner/axe-result-merger.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import axe from 'axe-core';
+import { AxeResultsReducer } from 'axe-result-converter';
+import { inject, injectable } from 'inversify';
+import { GlobalLogger } from 'logger';
+import { CombinedScanResultsProvider, CombinedScanResultsReadResponse } from 'service-library';
+import { CombinedScanResults } from 'storage-documents';
+
+@injectable()
+export class AxeResultMerger {
+    public constructor(
+        @inject(GlobalLogger) private readonly logger: GlobalLogger,
+        @inject(CombinedScanResultsProvider) protected readonly combinedScanResultsProvider: CombinedScanResultsProvider,
+        @inject(AxeResultsReducer) protected readonly axeResultsReducer: AxeResultsReducer,
+    ) {}
+
+    public async mergeAxeResults(axeScanResults: axe.AxeResults, combinedResultsBlobId: string): Promise<CombinedScanResults> {
+        const blobReadResponse = await this.getOrCreateCombinedResultsBlob(combinedResultsBlobId);
+        const combinedScanResults = blobReadResponse.results;
+
+        combinedScanResults.urlCount.total++;
+        if (axeScanResults.violations.length > 0) {
+            combinedScanResults.urlCount.failed++;
+        } else {
+            combinedScanResults.urlCount.passed++;
+        }
+
+        this.axeResultsReducer.reduce(combinedScanResults.axeResults, axeScanResults);
+        const blobWriteResponse = await this.combinedScanResultsProvider.writeCombinedResults(
+            combinedResultsBlobId,
```

Comment:
if an undefined blob Id is passed in, we'll need to generate a new one somewhere before writing (and return the new blob id so it can be updated in the WebsiteScanResult document)